testdata/default/cheats/AttachDelegation.t.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.18;
+
+import "ds-test/test.sol";
+import "cheats/Vm.sol";
+
+contract AttachDelegationTest is DSTest {
+    Vm constant vm = Vm(HEVM_ADDRESS);
+    uint256 pk = 77814517325470205911140941194401928579557062014761831930645393041380819009408;
+    address public ACCOUNT_A = 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266;
+
+    function testCreateDelegation(address implementation, uint64 nonce) public {
+        bytes32 delegation = vm.createDelegation(implementation, nonce);
+        (uint8 v, bytes32 r, bytes32 s) = vm.signDelegation(delegation, pk);
+
+        vm.attachDelegation(implementation, nonce, v, r, s);
```

Comment:
I believe delegations would be attached to separate transactions vs broadcast blocks, so it would be something like 
```solidity
vm.startBroadcast();
vm.attachDelegation(implementation, nonce, v, r, s);
// this transactions would be broadcasted along with attached 7702 delegation
token.transfer(....);
```

Reply:
thanks for the comment! looking at this closer - since 7702 makes the EOA itself the target of the tx, i think we need to route all calls through the EOA first. 

right now we can do this with a raw call:
```solidity
vm.startBroadcast();
vm.attachDelegation(implementation, nonce, v, r, s);
alice.call(abi.encodeCall(implementation.execute, (calls)));
```
but for better UX, maybe we could either:

1. change attachDelegation to take authority instead of implementation (since we have the implementation in the call itself):
```solidity
vm.attachDelegation(alice, nonce, v, r, s);
implementation.execute(calls);
```
2. or add a helper like:
```solidity
vm.delegatedCall(alice, implementation.execute, calls);
```
what do you think would be more ergonomic for users?